packages/client/hmi-client/src/components/models/tera-model.vue

```diff
@@ -386,6 +380,124 @@ let renderer: PetrinetRenderer | null = null;
 let eventX = 0;
 let eventY = 0;
 
+const modelConfigNames = ref([{ name: 'Config 1' }]);
+const fakeExtractions = ref(['Resource 1', 'Resource 2', 'Resource 3']);
+
+const initialValues = ref<StringValueMap[]>([{}]);
+const parameterValues = ref<StringValueMap[]>([{}]);
+const openValueConfig = ref(false);
+const cellValueToEdit = ref({ data: {}, field: '', index: 0 });
+
+const selectedModelConfig = ref();
+const openedWorkflowNodeStore = useOpenedWorkflowNodeStore();
+
+const modelConfiguration = computed(() => {
+	const newModelConfiguration: any[] = [];
+
+	for (let i = 0; i < modelConfigNames.value.length; i++) {
+		newModelConfiguration.push({
+			name: modelConfigNames.value[i].name,
+			...initialValues.value[i],
+			...parameterValues.value[i]
+		});
+	}
+
+	return newModelConfiguration;
+});
+
+function addModelConfiguration() {
+	modelConfigNames.value.push({ name: `Config ${modelConfigNames.value.length + 1}` });
+	initialValues.value.push(cloneDeep(initialValues.value[initialValues.value.length - 1]));
+	parameterValues.value.push(cloneDeep(parameterValues.value[parameterValues.value.length - 1]));
+}
+
+function addConfigValue() {
+	fakeExtractions.value.push(`Resource ${fakeExtractions.value.length + 1}`);
+}
+
+const onCellEditComplete = (event) => {
+	const { data, newValue, field } = event;
+
+	switch (field) {
+		case 'name':
+			data[field] = newValue;
+			break;
+		default:
+			break;
+	}
+};
+
+const onCellEditStart = (event) => {
+	const { data, field, index } = event;
+
+	if (field !== 'name') {
+		openValueConfig.value = true;
+		cellValueToEdit.value = { data, field, index };
+	}
+};
+
+function updateModelConfigValue() {
+	const { data, field, index } = cellValueToEdit.value;
+
+	if (initialValues.value[index][field]) {
+		initialValues.value[index][field] = data[field];
+	} else if (parameterValues.value[index][field]) {
+		parameterValues.value[index][field] = data[field];
+	}
+
+	openValueConfig.value = false;
+}
+
+function generateModelConfigValues() {
+	// Sync with workflow
+	if (
+		model.value &&
+		openedWorkflowNodeStore.assetId === model.value.id.toString() &&
+		openedWorkflowNodeStore.initialValues !== null &&
+		openedWorkflowNodeStore.parameterValues !== null
+	) {
+		// Shallow copy
+		initialValues.value = openedWorkflowNodeStore.initialValues as any; // Not sure why the typing doesn't match
```

Comment:
For typing, try defining `const initialValues = ref<StringValueMap[] | null>([{}]);`

Reply:
That makes it want to check if `initialValues` is null everywhere lol. I'll see if changing things to just a map can yield anything